lib/api/core/validation/types/geoShape.js

```diff
@@ -1,196 +1,223 @@
-var
-  util = require('util'),
-  BaseConstructor = require('../baseType'),
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2017 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const
+  BaseType = require('../baseType'),
   allowedShapeProperties = ['type', 'coordinates', 'radius', 'orientation', 'geometries'],
   allowedOrientations = ['right', 'ccw', 'counterclockwise', 'left', 'cw', 'clockwise'],
-  mutliTypes = ['multipoint', 'multilinestring', 'multipolygon'],
+  multiTypes = ['multipoint', 'multilinestring', 'multipolygon'],
   convertDistance = require('../../../dsl/util/convertDistance'),
   allowedShapeTypes = ['point', 'linestring', 'polygon', 'multipoint', 'multilinestring', 'multipolygon', 'geometrycollection', 'envelope', 'circle'];
 
 /**
- * @constructor
+ * @class GeoShapeType
  */
-function GeoShapeType () {
-  this.typeName = 'geo_shape';
-  this.allowChildren = false;
-  this.allowedTypeOptions = ['shapeTypes'];
-}
-
-util.inherits(GeoShapeType, BaseConstructor);
-
-/**
- * @param {TypeOptions} typeOptions
- * @param {GeoShape} fieldValue
- * @param {string[]} errorMessages
- */
-GeoShapeType.prototype.validate = function geoShapeValidate (typeOptions, fieldValue, errorMessages) {
-  return this.recursiveShapeValidation(typeOptions.shapeTypes, fieldValue, errorMessages);
-};
+class GeoShapeType extends BaseType {
+  constructor() {
+    super();
+    this.typeName = 'geo_shape';
+    this.allowChildren = false;
+    this.allowedTypeOptions = ['shapeTypes'];
+  }
 
-/**
- * @param {string[]} allowedShapes
- * @param {GeoShape} shape
- * @param {string[]} errorMessages
- * @returns {boolean}
- */
-GeoShapeType.prototype.recursiveShapeValidation = function geoShapeRecusriveShapeValidation (allowedShapes, shape, errorMessages) {
-  var
-    isMulti = mutliTypes.indexOf(shape.type) !== -1,
-    coordinateValidation,
-    result = true,
-    i;
-
-  if (!this.checkStructure(allowedShapes, shape, errorMessages)) {
-    return false;
+  /**
+   * @param {TypeOptions} typeOptions
+   * @param {GeoShape} fieldValue
+   * @param {string[]} errorMessages
+   */
+  validate(typeOptions, fieldValue, errorMessages) {
+    return this.recursiveShapeValidation(typeOptions.shapeTypes, fieldValue, errorMessages);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 92 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)